wasm/wat/convert.go

```diff
@@ -0,0 +1,66 @@
+package wat
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/tetratelabs/wazero/wasm"
+)
+
+// TextToBinary parses the configured source into a wasm.Module. This function returns when the source is exhausted or
+// an error occurs.
+//
+// Here's a description of the return values:
+// * result is the module parsed or nil on error
+// * err is a FormatError invoking the parser, dangling block comments or unexpected characters.
+func TextToBinary(source []byte) (result *wasm.Module, err error) {
+	var m *module
+	m, err = parseModule(source)
+	if err != nil {
+		return nil, err
```

Comment:
I think it is a good practice telling what were you trying to accomplish when the error happened e.g. `fmt.Errorf("failed to parse module: %v", err)` otherwise errors would look so random.

Reply:
parseModule already says this, so I think we are good. ex `1:30: redundant funcidx: $main in module.start`

Reply:
what I mean to say is that if we wrap we will lose the format and not gain anything in this case, but I totally agree we should think about error context and if it has it already or not. thanks for making me think

Reply:
ok! I added a comment. hopefully it works or make new line comments por favor!